plugins/modules/btrfs_subvolume.py

```diff
@@ -0,0 +1,672 @@
+        required: false
+        type: bool
+        default: false
+    default:
+        description:
+        - Make the targetted subvolume the default subvolume
+        required: false
+        type: bool
+        default: false
+    filesystem_device:
+        description:
+        - A block device contained within the btrfs filesystem to be targeted
+        - Useful when multiple btrfs filesystems are present to specify which filesystem should be targeted.
+        required: false
+        type: path
+    filesystem_label:
+        description:
+        - A descriptive label assigned to the btrfs filesystem to be targeted
+        - Useful when multiple btrfs filesystems are present to specify which filesystem should be targeted.
+        required: false
+        type: str
+    filesystem_uuid:
+        description:
+        - A unique identifier assigned to the btrfs filesystem to be targeted
+        - Useful when multiple btrfs filesystems are present to specify which filesystem should be targeted.
+        required: false
+        type: str
+    name:
+        description:
+        - Name of the subvolume/snapshot to be created
+        required: true
+        type: str
+    recursive:
+        description:
+        - When true, indicates that parent/child subvolumes should be created/removed
+        - as necessary to complete the operation (for state present/absent respectively).
+        required: false
+        type: bool
+        default: false
+    snapshot_source:
+        description:
+        - Source subvolume for the created snapshot.
+        - Infers that the created subvolume is a snapshot.
+        required: false
+        type: str
+    snapshot_conflict:
+        description:
+        - Policy defining behavior when a subvolume already exists at the path of the requested snapshot
+        - C(skip) - Create a snapshot only if a subvolume does not yet exist at the target location, otherwise indicate that no change is required.
+          Warning, this option does not yet verify that the target subvolume was generated from a snapshot of the requested source.
+        - C(clobber) - If a subvolume already exists at the requested location, delete it first.
+          This option is not idempotent and will result in a new snapshot being generated on every execution.
+        - C(error) - If a subvolume already exists at the requested location, return an error.
+          This option is not idempotent and will result in an error on replay of the module.
+        type: str
+        required: false
+        choices: [ skip, clobber, error ]
+        default: skip
+    state:
+        description:
+            - Whether the subvolume should exist or not
+        type: str
+        required: false
+        choices: [ absent, present ]
+        default: present
+
+notes:
+  - If any or all of the filesystem_device, filesystem_label or filesystem_uuid parameters are provided, there is expected
```

Comment:
```suggestion
  - If any or all of the I(filesystem_device), I(filesystem_label), or I(filesystem_uuid) parameters are provided, there is expected
```
Check out https://docs.ansible.com/ansible/devel/dev_guide/developing_modules_documenting.html#linking-and-other-format-macros-within-module-documentation for how to correctly format documentation.